### PR TITLE
Refactor Dockerfile, test scripts and CI

### PR DESCRIPTION
**Pull request by ambrussimon**
Supersedes https://github.com/flywheel-io/core/pull/1101

* Using @ehlertjd's rebase + travis.yml (encryption) cleanup
* Fixes according to review
   * Mongo ver + entrypoint err msg
   * Travis after_success `if` cleanup
   * Travis push for this branch as `scitran/core:refactor` for now
* Further limit test noise by only showing core log tail on integration test error
* Fix fly/fly compatibility, as this should be non-breaking

### Review Checklist

- Tests were added to cover all code changes
- Documentation was added / updated
- Code and tests follow standards in CONTRIBUTING.md

### CODE REVIEW

File: tests/README.md [hunk 1]
```diff
@@ -0,0 +1,24 @@
+## Testing
+Build scitran-core and run automated tests in a docker container:
```
Comment (ehlertjd) [inline tests/README.md]: `flywheel-core`?

**Comment by gsfr** [comment]
Great work, @ambrussimon.

@nagem @ryansanford Please support us in getting this landed this week!

File: .travis.yml [hunk 2]
```diff
@@ -1,33 +1,64 @@
...
+      git config --global user.name "Travis CI";
+      git config --global push.default simple;
+
+      if [ "$TRAVIS_EVENT_TYPE" == "push" ]; then
+          ./bin/push-docs.sh "$GIT_REMOTE" branches "$TRAVIS_BRANCH" "Travis Core Docs Build - $TRAVIS_BUILD_NUMBER";
+      else
+          ./bin/push-docs.sh "$GIT_REMOTE" tags "$TRAVIS_TAG" "Travis Core Docs Build - $TRAVIS_BUILD_NUMBER";
+      fi
+  fi
+
```
Comment (nagem) [inline .travis.yml]: Lots of references below to scitran/core, not sure has to be cleaned up now that we're on flywheel/core

File: tests/bin/tests.sh [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+    py.test --exitfirst tests/integration_tests/python $PYTEST_ARGS
+    rm tests/running_integration
+
+    log "INFO: Running pylint ..."
+    # TODO Enable Refactor and Convention reports
+    # TODO Move --disable into rc
+    pylint --rcfile=tests/.pylintrc --jobs=4 --reports=no --disable=C,R,W0312,W0141,W0110 api
+
+    # log "Running pep8 ..."
+    # pep8 --max-line-length=150 --ignore=E402 api
```
Comment (nagem) [inline tests/bin/tests.sh]: I like your aspirations!

**Comment by nagem** [comment]
Changes look good to me. Is addressing the existing references to `scitran` out of scope for this PR?

**Comment by gsfr** [comment]
Good catch, @nagem. Yes, we need to be pushing images to flywheel/core on docker hub, not to scitran/core.

**Comment by nagem** [comment]
For less important `scitran` cleanup, I opened this issue: https://github.com/flywheel-io/core/issues/1115

File: tests/README.md [hunk 4]
```diff
@@ -0,0 +1,24 @@
+## Testing
```
Comment (ryansanford) [inline tests/README.md]: Main README.md link to here needs to be updated

File: .dockerignore [hunk 5]
```diff
@@ -1,15 +1,11 @@
...
-.cache
-.coverage*
-coverage.xml
-htmlcov
-node_modules
+!api
+!bin
+!docker/uwsgi-config.ini
+!docker/uwsgi-entrypoint.sh
+!requirements.txt
```
Comment (ryansanford) [inline .dockerignore]: `docker/bootstrap-accounts.sh` not existing in the docker image is a breaking change. Add an exception for that file also.

File: Dockerfile [hunk 6]
```diff
@@ -1,103 +1,90 @@
...
+    && chmod +x /usr/local/bin/gosu \
+    && rm -rf "$GNUPGHOME" /tmp/gosu.asc \
+    && mkdir -p \
+        /var/scitran/code/api \
+        /var/scitran/config \
+        /var/scitran/data \
+        /var/scitran/keys \
+        /var/scitran/logs
+
+VOLUME ["/var/scitran/keys", "/var/scitran/data", "/var/scitran/logs"]
```
Comment (ryansanford) [inline Dockerfile]: @gsfr @ambrussimon  This comment is not a blocker. Just want you to be aware that adjacent VOLUME or other commands that don't alter the image filesystem no longer result in multiple image layers, so there is no longer any efficiency gain with using this squashed syntax.

**Comment by gsfr** [comment]
🎉

**Pull request merged**